src/scratch/ScratchRuntime.as

```diff
@@ -89,49 +145,342 @@ public class ScratchRuntime {
 		processEdgeTriggeredHats();
 		interp.stepThreads();
 		app.stagePane.commitPenStrokes();
+		
+		if (ready>=0) {
+			app.stagePane.countdown(count);
+		}
 	}
 
-//-------- recording test ---------
+//-------- recording video code ---------
 	public var recording:Boolean;
 	private var frames:Array = [];
-
+	private var sounds:Array = [];
+	private var recTimer:Timer;
+	private var baFlvEncoder:ByteArrayFlvEncoder;
+	private var position:int;
+	private var secs:Number;
+	private var alreadyDone:int;
+	
+	private var pSound:Boolean;
+	private var mSound:Boolean;
+	private var mouse:Boolean;
+	private var cursor:Boolean;
+	public var fullEditor:Boolean;
+	private var framerate:Number;
+	private var width:int;
+	private var height:int;
+	public var ready:int=-1;
```

Comment:
I think it would make it easier to understand and maintain this code if you could rename and/or add comments to some of these variables to clarify their meaning to someone who isn't familiar with the video code. I think this is especially important since these variables are in a class that isn't just about video recording.

For example, it might make sense to rename `position` to `videoPosition` or something like that. In the case of `ready` (and maybe others?) it could also make sense to turn it into a pseudo-`enum`, similar to `LogLevel`, `MouseEvent`, and so on, to express the meaning of the individual values.